pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -305,28 +305,36 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) (Results, err
 		if !ok {
 			break
 		}
-
 		// Schedule to existing nodes or create a new node
-		if podErrors[pod] = s.add(ctx, pod); podErrors[pod] == nil {
-			delete(podErrors, pod)
+		p := pod.DeepCopy()
+		for {
+			if podErrors[pod] = s.add(ctx, pod); podErrors[pod] == nil {
+				delete(podErrors, pod)
+				break
+			}
+			if IsReservedOfferingError(podErrors[pod]) {
```

Comment:
Can we retain the previous comment for why we don't relax with reserved offerings?